src/history/hs_rec.c

```diff
@@ -711,14 +690,14 @@ __wt_hs_delete_key_from_ts(WT_SESSION_IMPL *session, WT_CURSOR *hs_cursor, uint3
 }
 
 /*
- * __hs_fixup_out_of_order_from_pos --
- *     Fixup existing out-of-order updates in the history store. This function works by looking
- *     ahead of the current cursor position for entries for the same key, removing them and
- *     reinserting them at the timestamp that is currently being inserted.
+ * __hs_delete_reinsert_from_pos --
+ *     Delete updates in the history store if they are larger or equal to the specified timestamp
+ *     and reinsert them with ts-1 timestamp. This function works by looking ahead of the current
```

Comment:
Again the idea of reinserting at ts-1 is bit confusing. Maybe we should provide the new timestamp explicitly.

Reply:
Delete updates in the history store if they are -> Delete updates in the history store if the start timestamp of the update is.

reinsert them with ts-1 timestamp -> and optionally reinsert them with ts-1 timestamp